File: 0001-two-sum/0001-two-sum.cpp
class Solution
{
    public:
        vector<int> twoSum(vector<int> &nums, int target)
        {
            int i, j;
            vector<int> v;
            int n=nums.size();
            map<int,int> m;
            for(int i=0;i<n;i++)
                m[nums[i]]=i;
            for(i=0;i<n;i++)
            {
                if(m.find(target-nums[i])!=m.end() && i != m[target-nums[i]])
                    return{i,m[target-nums[i]]};
            }
            return v;
        }
};

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    mp := make(map[int]int)
    
    for i,t:=range nums{
        j, exist := mp[target-t]
        if exist{
            return []int{i, j}
        }
        mp[t]=i
    }
    return []int{}
}